cpp/include/cudf/copying.hpp

```diff
@@ -717,5 +717,47 @@ std::unique_ptr<scalar> get_element(
   size_type index,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Indicates whether a row can be sampled more than once.
+ **/
+enum class row_multi_sampling : bool {
+  DISALLOWED,  // A row can be sampled only once
+  ALLOWED      // A row can be sampled more than once
+};
+
+/**
+ * @brief Gather `n` samples from given `input_table` randomly
+ *
+ * @code{.pseudo}
+ * Example:
+ * input: {col1: {1, 2, 3, 4, 5}, col2: {6, 7, 8, 9, 10}}
+ * n: 3
+ * multi_smpl: false
+ *
+ * output:       {col1: {3, 1, 4}, col2: {8, 6, 9}}
+ *
+ * multi_smpl: true
+ *
+ * output:       {col1: {3, 1, 1}, col2: {8, 6, 6}}
+ * @endcode
+ *
+ * @throws cudf::logic_error if `n` > `input.num_rows()` and `multi_smpl` == DISALLOWED.
+ * @throws cudf::logic_error if `n` < 0.
+ *
+ * @param input View of a table to sample.
+ * @param n non-negative number of samples expected from `input`.
+ * @param multi_smpl Allow or disallow sampling of the same row more than once.
+ * @param seed Seed value to initiate random number generator.
+ * @param mr Device memory resource used to allocate the returned table's device memory
+ *
+ * @return std::unique_ptr<table> Table containing samples from `input`
+ */
+std::unique_ptr<table> sample(
+  table_view const& input,
+  size_type const n,
+  row_multi_sampling multi_smpl       = row_multi_sampling::DISALLOWED,
```

Comment:
```suggestion
  row_multi_sampling multi_sampling   = row_multi_sampling::DISALLOWED,
```
and corresponding references.